Content.Server/Xenoarchaeology/XenoArtifacts/ArtifactComponent.cs

```diff
@@ -1,39 +1,128 @@
+using Content.Shared.Xenoarchaeology.XenoArtifacts;
+using Robust.Shared.Serialization.TypeSerializers.Implementations;
+
 namespace Content.Server.Xenoarchaeology.XenoArtifacts;
 
 [RegisterComponent]
 public sealed class ArtifactComponent : Component
 {
     /// <summary>
-    ///     Should artifact pick a random trigger on startup?
+    /// The artifact's node tree.
+    /// </summary>
+    [ViewVariables]
+    public ArtifactTree? NodeTree;
+
+    /// <summary>
+    /// The current node the artifact is on.
+    /// </summary>
+    [ViewVariables]
+    public ArtifactNode? CurrentNode;
+
+    #region Node Tree Gen
+    /// <summary>
+    /// Minimum number of nodes to generate, inclusive
     /// </summary>
-    [DataField("randomTrigger")]
-    public bool RandomTrigger = true;
+    [DataField("nodesMin")]
+    public int NodesMin = 3;
 
     /// <summary>
-    ///     List of all possible triggers activations.
-    ///     Should be same as components names.
+    /// Maximum number of nodes to generate, exclusive
     /// </summary>
-    [DataField("possibleTriggers")]
-    public string[] PossibleTriggers = {
-        "ArtifactInteractionTrigger",
-        "ArtifactGasTrigger",
-        "ArtifactHeatTrigger",
-        "ArtifactElectricityTrigger",
-    };
+    [DataField("nodesMax")]
+    public int NodesMax = 9;
+    #endregion
 
     /// <summary>
-    ///     Cooldown time between artifact activations (in seconds).
+    /// Cooldown time between artifact activations (in seconds).
     /// </summary>
-    [DataField("timer")]
+    [DataField("timer", customTypeSerializer: typeof(TimespanSerializer))]
     [ViewVariables(VVAccess.ReadWrite)]
-    public double CooldownTime = 10;
+    public TimeSpan CooldownTime = TimeSpan.FromSeconds(5);
 
     /// <summary>
-    ///     Is this artifact under some suppression device?
-    ///     If true, will ignore all trigger activations attempts.
+    /// Is this artifact under some suppression device?
+    /// f true, will ignore all trigger activations attempts.
     /// </summary>
     [ViewVariables(VVAccess.ReadWrite)]
     public bool IsSuppressed;
 
+    /// <summary>
+    /// The last time the artifact was activated.
+    /// </summary>
+    [DataField("lastActivationTime", customTypeSerializer: typeof(TimespanSerializer))]
     public TimeSpan LastActivationTime;
 }
+
+/// <summary>
+/// A tree of nodes.
+/// </summary>
+[DataDefinition]
+public sealed class ArtifactTree
```

Comment:
i guess one thing to note is that i envisioned this being a cyclic graph structure rather than a tree, so you'd be able to traverse anywhere in the artifact graph rather than just descending. although i dont have any strong attachment to that concept of it over this, so i'm ok with having it be a tree

Reply:
ANOTHER thing to note is that i kinda envisioned the separate nodes of the artifact being entirely separate entities (in the abstract sense, not necessarily literally), so the effects of one would get 'undone' when you traverse to a diff node but ALSO i dont have any super strong attachment to that and im fine with this approach too